build-logic/packaging/src/main/kotlin/gradlebuild/packaging/tasks/ExtractJavaAbi.kt

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild.packaging.tasks
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.provider.SetProperty
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Classpath
+import org.gradle.api.tasks.CompileClasspath
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+import org.gradle.internal.tools.api.ApiClassExtractor
+import org.gradle.internal.tools.api.impl.JavaApiMemberWriter
+import org.gradle.workers.WorkAction
+import org.gradle.workers.WorkParameters
+import org.gradle.workers.WorkerExecutor
+import java.io.File
+import java.io.InputStream
+import javax.inject.Inject
+
+
+/**
+ * Extract API-only classes from classes directories.
+ *
+ * Keeps only the following:
+ *
+ * -  API stubs of public classes in the specified packages
+ * - `META-INF/groovy/org.codehaus.groovy.runtime.ExtensionModule`
+ * - `META-INF/services/org.codehaus.groovy.transform.ASTTransformation`
+ * - `META-INF/\*.kotlin_module`
+ */
+@CacheableTask
+abstract class ExtractJavaAbi : DefaultTask() {
+
+    @get:Input
+    abstract val packages: SetProperty<String>
+
+    @get:CompileClasspath
+    abstract val classesDirectories: ConfigurableFileCollection
+
+    @get:OutputDirectory
+    abstract val outputDirectory: DirectoryProperty
+
+    @get:Classpath
+    abstract val extractorClasspath: ConfigurableFileCollection
+
+    @get:Inject
+    protected abstract val workerExecutor: WorkerExecutor
+
+    @TaskAction
+    fun execute() {
+        // Run using classloader isolation to avoid using the ABI extractor from the runtime Gradle distribution
```

Comment:
💅 Should this be:
```suggestion
        // Run using process isolation to avoid using the ABI extractor from the runtime Gradle distribution
```
?